core/field_dropdown.js

```diff
@@ -413,17 +413,20 @@ Blockly.FieldDropdown.prototype.setValue = function(newValue) {
       var content = options[i][0];
       if (typeof content == 'object') {
         this.imageJson_ = content;
-        this.setText(content.alt);
+        this.text_ = content.alt;
```

Comment:
Why did you change these away from setText?

Reply:
Because if the text hasn't changed setText won't do the rerender, so I would need to rerender here.  But if the text had changed, setText would do a rerender and I wouldn't know (except by storing the old and new values here and comparing them), so forcing a rerender would cause an extraneous rendering pass.  I didn't want to change setText to return whether it updated, since that would be a change on Field.  So I'm just doing the actual effects of setText that I care about.

Reply:
Okay.  More evidence that we need to clean up fields and setText/setValue!  But, I agree that this bug is worth fixing and this is probably an okay work around for now.  It might cause an awkward conflict in scratch-blocks so we should watch for that.